### DISCUSSION TITLE
onSegmentChange event fires over and over

### DISCUSSION THREAD

**Discussion by Lex-Kent** [speechly/speechly#307] [Q&A]
Hi all,

I'm having a weird issue with a very simple application which listens for a date in the future. My config looks like this:

![image](https://github.com/speechly/speechly/assets/107930534/9c7cc411-dbae-4563-bc72-21588f6097c8)

and it works really well (with the exception of some words such as "immediately", which doesn't return today's date)

but when the user utters something which ends with the year spoken aloud, the `onSegmentChange` event fires a few hundred times after they have finished speaking, before `isFinal` changes to `true` and my app continues the conversation. This leads to a very long pause for no reason I can ascertain.

A typical problem phrase is "**_March twenty twenty four_**".

Can anyone help shed some light please?

Thanks so much

**Comment by teelisyys** [comment]
Hi @Lex-Kent ! Thank you for using Speechly!

This is indeed a bug.

Unfortunately it is caused by unexpected behaviour of the base ASR model in certain edge cases related to the text-adaptation. We expect this issue to be mitigated in future updates.

At this time, there is no generally applicable workaround available.
